android/src/main/java/io/radar/capacitor/RadarPlugin.java

```diff
@@ -841,4 +849,54 @@ private static String[] stringArrayForJSArray(JSArray jsArr) {
         }
     }
 
+    private RadarTrackingOptions trackingOptionsFromCapacitorJson(JSONObject json) {
+        RadarTrackingOptions options = RadarTrackingOptions.fromJson(json);
+
+        Date startTrackingAfter = dateFromJSONDateField(json, "startTrackingAfter");
+
+        if (startTrackingAfter != null) {
+            options.setStartTrackingAfter(startTrackingAfter);
+        }
+
+        Date stopTrackingAfter = dateFromJSONDateField(json, "stopTrackingAfter");
+
+        if (stopTrackingAfter != null) {
+            options.setStopTrackingAfter(stopTrackingAfter);
+        }
+
+        return options;
+    }
+
+    private RadarTripOptions tripOptionsFromCapacitorJson(JSONObject json) {
+        RadarTripOptions options = RadarTripOptions.fromJson(json);
+
+        Date scheduledArrivalAt = dateFromJSONDateField(json, "scheduledArrivalAt");
+
+        if (scheduledArrivalAt != null) {
+            options.setScheduledArrivalAt(scheduledArrivalAt);
+        }
+
+        return options;
+    }
+
+    /* Date Utilities */
+    private DateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+    private Date dateFromJSONDateField(JSONObject json, String fieldName) {
+        if (json.has(fieldName)) {
+            try {
+                String dateString = json.getString(fieldName);
```

Comment:
Can we possibly use this:
```
The CAPPluginCall convenience accessor getDate will handle both data types and return a Date object.
```
 
from the docs (https://capacitorjs.com/docs/core-apis/data-types#dates)

Reply:
Excellent question! `CAPPluginCall.getDate()` _would_ be perfect, but only for top-level arguments. In our case, dates like `startTrackingAfter` and `stopTrackingAfter` are properties of the tracking options that are passed in, and we take that tracking options dictionary and pass it verbatim to our native `RadarTrackingOptions.from(:)` call.